libs/agno/agno/workflow/v2/step.py

```diff
@@ -666,55 +666,53 @@ async def aexecute_stream(
                     else:
                         raise e
 
-    def _parse_message_data(self, message_data: Optional[Union[BaseModel, Dict[str, Any]]]) -> Optional[str]:
-        """Parse the message data into a string"""
-        data_str = None
-        if message_data is not None:
-            if isinstance(message_data, BaseModel):
-                data_str = message_data.model_dump_json(indent=2, exclude_none=True)
-            elif isinstance(message_data, dict):
-                import json
-
-                data_str = json.dumps(message_data, indent=2, default=str)
-            else:
-                data_str = str(message_data)
-        return data_str
+    def _convert_input_to_string(
+        self, message: Optional[Union[str, Dict[str, Any], List[Any], BaseModel]]
+    ) -> Optional[str]:
+        """Convert various input types to string"""
+        if message is None:
+            return None
+        if isinstance(message, str):
+            return message
+        elif isinstance(message, BaseModel):
+            return f"Process the following structured data:\n{message.model_dump_json(indent=2, exclude_none=True)}"
```

Comment:
We shouldn't do this here.
If the step executor is a function, I would expect the StepInput.message to be the actual message, not converted to string.
If the step executor is agent/team, same thing. We should update agent/team to handle BaseModel as a message type